crates/relayer/src/event/bus.rs

```diff
@@ -14,34 +14,22 @@ impl<T> Default for EventBus<T> {
 
 impl<T> EventBus<T> {
     pub fn new() -> Self {
-        Self {
-            txs: VecDeque::new(),
-        }
+        Self { txs: Vec::new() }
     }
 
     pub fn subscribe(&mut self) -> channel::Receiver<T> {
         let (tx, rx) = channel::unbounded();
-        self.txs.push_back(tx);
+        self.txs.push(tx);
         rx
     }
 
     pub fn broadcast(&mut self, value: T)
     where
         T: Clone,
     {
-        let mut disconnected = Vec::new();
-
-        for (idx, tx) in self.txs.iter().enumerate() {
-            // TODO: Avoid cloning when sending to last subscriber
-            if let Err(channel::SendError(_)) = tx.send(value.clone()) {
-                disconnected.push(idx);
-            }
-        }
-
-        // Remove all disconnected subscribers
-        for idx in disconnected {
-            self.txs.remove(idx);
```

Comment:
When there are multiple disconnected receivers, indexes of the latter ones will change after the first receiver is deleted.

Reply:
that's indeed a bug in official Hermes repo